### DISCUSSION TITLE
How to use the new HTTP Remote Control

### DISCUSSION THREAD

**Discussion by nugatritter** [bitfocus/companion#2662] [Q&A] [answered]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I could press and release the button 99/0/4 with the old / deprecated HTTP remote control by 
`wget http://127.0.0.1:8000/press/bank/99/32`
I cannot find the way to do same thing with the new HTTP protocol. I tried following:
`wget http://127.0.0.1:8000/api/location/99/0/4/press`
and get as result
`Verbindungsaufbau zu 127.0.0.1:8000... verbunden.
HTTP Anforderung gesendet, warte auf Antwort... 200 OK
Länge: 1440 (1,4K) [text/html]
In »press.10« speichern.

100%[=============================================================================> ] 1.440       --.-K/s   in 0s

2023-11-26 16:11:24 (113 MB/s) - »press.10« gespeichert [1440/1440]`
The first rerquest works but seconds does nothing on Companion side.
I also tried to use wget --post-data '/api/location/99/0/4/press' http://127.0.0.1:8000
But I got only this error message.
SYSTEM_WGETRC = c:/progra~1/wget/etc/wgetrc
syswgetrc = C:\Program Files (x86)\GnuWin32/etc/wgetrc
--2023-11-26 16:19:53--  http://127.0.0.1:8000/
Verbindungsaufbau zu 127.0.0.1:8000... verbunden.
HTTP Anforderung gesendet, warte auf Antwort... 404 Not Found
2023-11-26 16:19:53 FEHLER 404: Not Found.

### Steps To Reproduce

1. Check Setting->HTTP->HTTP API and Deprecated HTTP API are activated
2. Set action of button 99/0/4 to "internal: System: Restart companion"
3. Go to command line and call wget http://127.0.0.1:8000/press/bank/99/32
4. You will see the short restart of Companion in the web interface
5. Go to command line and call wget http://127.0.0.1:8000/api/location/99/0/4/press
6. Companion will not restart

### Expected Behavior

wget http://127.0.0.1:8000/press/bank/99/32
does the same as
wget http://127.0.0.1:8000/api/location/99/0/4/press

### Environment (please complete the following information)

```markdown
- OS: Windows 11
- Browser: Firefox 120.0
- Companion Version: companion-win64-3.2.0+6552-beta-3b0356be and companion-win64-3.1.2+6243-stable-05c37a10
```

**Comment by Julusian** [comment, marked as answer]
Try `wget --post-data={} http://127.0.0.1:8000/api/location/99/0/4/press`

**Comment by nugatritter** [comment]
Many thanks! That works fine.
